book/src/articles/objects_deep_dive.md

```diff
@@ -0,0 +1,136 @@
+# Objects deep dive
+
+An object is a sprite drawn to an arbitrary part of the screen.
+They are typically used for anything that moves such as characters and NPCs.
+All objects can be flipped and affine objects can have an affine transformation applied to them that can rotate and scale them.
+
+
+## Importing sprites
+
+Sprites are imported from aseprite files.
+Aseprite is an excellent pixel sprite editor that can be acquired for low cost or compiled yourself for free.
+It also provides features around adding tags for grouping sprites to form animations and dictating how an animation is performed.
+This makes it very useful for creating art for use by agb.
+
+
+You can import 15 colour sprites using the `include_aseprite` macro.
+In a single invocation of `include_aseprite` the palettes are all optimised together.
+For example, you might use the following to import some sprites.
+
+```rust
+agb::include_aseprite(mod sprites, "sprites.aseprite", "other_sprites.aseprite");
+```
+
+You can import 255 colour sprites using the `include_aseprite_256` macro similarly.
+
+
+## ROM and VRAM
+
+Sprites must be in VRAM to be displayed on screen.
+The `SpriteVram` type represents a sprite in VRAM.
+This implements the `From` trait from `&'static Sprite`.
+The `From` implementation will deduplicate the sprites in VRAM, this means that you can repeatedly use the same sprite and it won't blow up your VRAM usage.
+This deduplication does have the performance implication of requiring a HashMap lookup, although for many games this will a rather small penalty.
+By storing and reusing the `SpriteVram` you can avoid this lookup.
+Furthermore, `SpriteVram` is reference counted, so `Clone`ing it is cheap and doesn't allocate more `VRAM`.
+
+
+```rust
+agb::include_aseprite(mod sprites, "examples/chicken.aseprite");
+
+let sprite = agb::display::object::SpriteVram = sprites::IDLE.sprite(0).into();
```

Comment:
It seems calling `SpriteVram::from(...)` is the more common way I've seen this written rather than using `.into()`